isthmus/src/test/java/io/substrait/isthmus/ApplyJoinPlanTest.java

```diff
@@ -0,0 +1,69 @@
+package io.substrait.isthmus;
+
+import org.apache.calcite.adapter.tpcds.TpcdsSchema;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.validate.SqlConformanceEnum;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ApplyJoinPlanTest {
+
+  @Test
+  public void lateralJoinQuery() {
+    TpcdsSchema schema = new TpcdsSchema(1.0);
+    String sql;
+    sql =
+        """
+        SELECT ss_sold_date_sk, ss_item_sk, ss_customer_sk
+        FROM store_sales CROSS JOIN LATERAL
+          (select i_item_sk from item where item.i_item_sk = store_sales.ss_item_sk)""";
+
+    SqlToSubstrait s = new SqlToSubstrait();
+    Assertions.assertThrows(
+        UnsupportedOperationException.class,
+        () -> s.execute(sql, "tpcds", schema),
+        "Lateral join is not supported");
```

Comment:
With this `not supported` exception as the expected result, it's hard to see how the code change works.

If Substrait's spec for APPLY is not ready yet, is it possible to verify the first step's output, instead of going through all the steps from SQL -> Substrait? Another choice is to wait until the spec of APPLY is ready.

Reply:
I agree. And I like the suggestion to expose internal data structures to validate the first step. I will try to create it.

Please note that the `Calcite-plan` for `APPLY` query already exists. As the Calcite-plan is external to Substrait, parsing it is deterministic and is independent of the `APPLY` spec in Substrait. As such, I believe this PR should precede calcite-to-Substrait. In-fact this was requested in one of the community meetings as it helps in understanding how `APPLY` is coded in Calcite.

But I agree, the tests in the PR do not confirm if the intended data structures in Substrait are correctly updated. I will try to incorporate it. Thanks for the suggestion.

Reply:
Totally agree.  It helps to verify the result of first step in the test, as it will verify that the internal data is updated as expected. Thanks for the effort.

Reply:
I have pushed a change to validate the first step (calcite plan parsing). I also added the calcite plan corresponding to the query for reference. Thanks.

Reply:
The new tests look great. I just have one minor comment. Thank you for responding with new code change.

Reply:
Thanks for suggesting it in the first place. I appreciate your thorough review.